test/Contract/Buy.purs

```diff
@@ -0,0 +1,80 @@
+module Test.Contract.Buy (suite) where
+
+import Contract.Prelude
+
+import Contract.Address
+  ( getNetworkId
+  , getWalletAddress
+  , ownPaymentPubKeyHash
+  , payPubKeyHashEnterpriseAddress
+  )
+import Contract.Monad (liftContractM, liftedM)
+import Contract.Numeric.Natural as Nat
+import Contract.Test.Plutip (runPlutipContract, withKeyWallet, withStakeKey)
+import Contract.Transaction (awaitTxConfirmed)
+import Data.BigInt as BigInt
+import Mote (only, test)
+import Seabug.Contract.MarketPlaceBuy (marketplaceBuy')
+import Seabug.MarketPlace (marketplaceValidatorAddr)
+import Test.Contract.Util
+  ( assertContract
+  , assertLovelaceDecAtAddr
+  , assertLovelaceIncAtAddr
+  , callMintCnft
+  , callMintSgNft
+  , checkNftAtAddress
+  , findUtxoWithNft
+  , mintParams1
+  , plutipConfig
+  , privateStakeKey
+  , withAssertions
+  )
+import TestM (TestPlanM)
+
+suite :: TestPlanM Unit
+suite =
+  only $ test "Buy" do
+    let
+      distribution =
+        ( withStakeKey privateStakeKey
+            [ BigInt.fromInt 1_000_000_000
+            , BigInt.fromInt 2_000_000_000
+            ]
+        ) /\
+          ( withStakeKey privateStakeKey
+              [ BigInt.fromInt 1_000_000_000
+              , BigInt.fromInt 2_000_000_000
+              ]
+          )
+    runPlutipContract plutipConfig distribution \(seller /\ buyer) -> do
+      networkId <- getNetworkId
+      sellerPayAddr <- withKeyWallet seller do
```

Comment:
`seller` is a KeyWallet, which has a few nice utils in it, like [this](https://plutonomicon.github.io/cardano-transaction-lib/Wallet.Key.html#v:keyWalletPrivatePaymentKey)
You can probably build this address without requiring `withKeyWallet`

Reply:
It could be updated to something like

```purescript
      sellerPayAddr <-
        liftEffect
          ( wrap <<< wrap <<< publicKeyHash <$>
              publicKeyFromPrivateKey
                (unwrap $ keyWalletPrivatePaymentKey seller)
          ) >>= liftContractM "Could not get seller payment address" <<<
          payPubKeyHashEnterpriseAddress networkId

```

But that requires importing from `Serialization` which afaik isn't part of the public interface. Do you think it would be a significant performance improvement?